span_implementation.go

```diff
@@ -2,23 +2,27 @@ package zipkin
 
 import (
 	"sync"
+	"sync/atomic"
 	"time"
+
+	"github.com/openzipkin/zipkin-go/model"
 )
 
 type spanImpl struct {
 	mtx sync.RWMutex
-	SpanModel
-	tracer    *Tracer
-	isSampled bool
+	model.SpanModel
+	tracer          *Tracer
+	isSampled       int32 // atomic bool (1 = true, 0 = false)
+	explicitContext bool
```

Comment:
what does this mean?

Reply:
explicitContext is used to identify if the SpanContext to be handled is explicitly set (think the use case where a worker process inject the required id's from env var) and thus should not change or if it is parent context that either needs to be joined to or used to create a child span. If explicitContext is set depends on which SpanOption is used. Parent will set it to false, WithSpanContext will set it to true.